components/higher-order/with-filters/test/index.js

```diff
@@ -1,35 +1,62 @@
 /**
  * External dependencies
  */
-import { mount, shallow } from 'enzyme';
+import { shallow } from 'enzyme';
+import TestUtils from 'react-dom/test-utils';
+import ReactDOM from 'react-dom';
 
 /**
  * WordPress dependencies
  */
 import { addFilter, removeAllFilters, removeFilter } from '@wordpress/hooks';
+import { Component } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import withFilters from '..';
 
+const assertExpectedHtml = ( wrapper, containerTag, expectedHTML ) => {
+	const element = TestUtils.findRenderedDOMComponentWithTag( wrapper, containerTag );
```

Comment:
Technically speaking, we should be searching for a root component in here. Did you consider using [https://reactjs.org/docs/test-utils.html#findrenderedcomponentwithtype](https://reactjs.org/docs/test-utils.html#findrenderedcomponentwithtype) instead?

Reply:
I'm not sure I follow why I want to search for the root component here.  Can you elaborate?`TestUtils.findRenderedComponentWithType` returns a component instance, what I want here is the DOM Element matching the given tag (`wrapper` is the root component which the given element is derived from).

Reply:
nvm I think I grasp now the reason for your comment.  It's to ensure the html output for the entire component matches the expectation.